### PR TITLE
Adds a cronjob which allows catalog to be refreshed

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/config/sync-catalog/32-sync-catalog-cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: catalog-refresh
+  namespace: tekton-hub
+  labels:
+    app: api
+spec:
+  schedule: "*/30 * * * *"
```
Comment (sthaha) [inline api/config/sync-catalog/32-sync-catalog-cronjob.yaml:10]: we need to maintain only the last 3/5 jobs, rest can be garbage collected like we do now in `experimental`

File: api/config/sync-catalog/32-sync-catalog-cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          containers:
+            - name: catalog-refresh
+              image: curlimages/curl:7.71.1
+              env:
+                - name: Token
```
Comment (SM43) [inline api/config/sync-catalog/32-sync-catalog-cronjob.yaml:19]: `Token` - > `AGENT_TOKEN`?

File: api/config/sync-catalog/32-sync-catalog-cronjob.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+---
```
Comment (SM43) [inline api/config/sync-catalog/32-sync-catalog-cronjob.yaml:1]: folder name `sync-catalog` -> `catalog refresh`? to maintain catalog refresh as consistent everywhere? WDYT?

**Review by SM43 — CHANGES_REQUESTED** [review]

File: api/config/sync-catalog/32-sync-catalog-cronjob.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+          containers:
+            - name: catalog-refresh
+              image: curlimages/curl:7.71.1
+              env:
+                - name: Token
+                  valueFrom:
+                    secretKeyRef:
+                      name: catalog-refresh
+                      key: Token
+              args: ["-i", "-X", "PUT", "-H", "Authorization:$(Token)", "api:8000/catalog/refresh"]
```
Comment (sthaha) [inline api/config/sync-catalog/32-sync-catalog-cronjob.yaml:24]: wouldn't 8000 fail?

Comment (SM43) [reply api/config/sync-catalog/32-sync-catalog-cronjob.yaml:24]: Why? API server runs on 8000.

Comment (sthaha) [reply api/config/sync-catalog/32-sync-catalog-cronjob.yaml:24]: @SM43 I thought the service is reachable on 80but it isn't...

**Review by sthaha — COMMENTED** [review]
/lgtm
/approve
/meow

**Pull request merged**